File: app/src/main/java/com/saienko/androidthings/barman/db/position/PositionDao.java
package com.saienko.androidthings.barman.db.position;

import android.arch.persistence.room.*;

import java.util.List;

/**
 * Created
 * User: Vasiliy Saienko
 * Date: 9/12/17
 * Time: 11:57
 */

@Dao
public interface PositionDao {

    @Query("SELECT * FROM Position")
    List<Position> getAll();

//    @Insert
//    void insertAll(Position... positions);

    @Insert
    void insert(Position position);

    @Delete
    void delete(Position position);

    @Query("DELETE FROM Position")
    void clear();

    @Update
    void update(Position position);

    @Query("SELECT * FROM Position WHERE motorId = :motorId")
    Position getByMotorId(long motorId);

    @Query("SELECT * FROM Position WHERE componentId = :componentId")
    Position getByComponentId(long componentId);

    @Query("DELETE FROM Position WHERE motorId = :motorId")
    void deleteByMotorId(long motorId);
}

File: app/src/main/java/com/saienko/androidthings/barman/db/dao/PositionDao.kt
package com.saienko.androidthings.barman.db.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Query
import com.saienko.androidthings.barman.db.entity.Position

/**
 * Created
 * User: Vasiliy Saienko
 * Date: 9/12/17
 * Time: 11:57
 */

@Dao
interface PositionDao : BaseDao<Position> {

    @get:Query("SELECT * FROM Position")
    val all: MutableList<Position>

    @Query("DELETE FROM Position WHERE id = :id")
    fun delete(id: Long?)

    @Query("DELETE FROM Position")
    fun clear()

    @Query("SELECT * FROM Position WHERE motorId = :motorId")
    fun getByMotorId(motorId: Long): Position

    @Query("DELETE FROM Position WHERE motorId = :motorId")
    fun deleteByMotorId(motorId: Long)

    @Query("SELECT * FROM Position WHERE componentId = :componentId")
    fun getByComponentId(componentId: Long): Position

    @Query("SELECT * FROM Position WHERE id = :positionId")
    fun get(positionId: Long): Position
}
